src/gennylib/include/gennylib/topology.hpp

```diff
@@ -0,0 +1,219 @@
+// Copyright 2021-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef HEADER_6D7AD2AF_7CD1_49E4_9712_741ABD354DFC_INCLUDED
+#define HEADER_6D7AD2AF_7CD1_49E4_9712_741ABD354DFC_INCLUDED
+
+#include <vector>
+#include <variant>
+#include <mutex>
+#include <sstream>
+
+#include <gennylib/v1/PoolFactory.hpp>
+#include <gennylib/service.hpp>
+
+namespace genny {
+
+class AbstractTopologyDescription;
```

Comment:
super nit: Just call this `Topology` - the fact that it's abstract is its business, and "description" doesn't add much to the name. Push back if you disagree no biggie.